airbyte-workers/src/main/java/io/airbyte/workers/normalization/NormalizationAirbyteStreamFactory.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.normalization;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.logging.MdcScope;
+import io.airbyte.protocol.models.AirbyteLogMessage;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.workers.internal.AirbyteStreamFactory;
+import java.io.BufferedReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates a stream from an input stream. The produced stream attempts to parse each line of the
+ * InputStream into a AirbyteMessage. If the line cannot be parsed into a AirbyteMessage it is
+ * assumed to be from dbt. dbt [error] messages are also parsed
+ *
+ * <p>
+ * If a line starts with a AirbyteMessage and then has other characters after it, that
+ * AirbyteMessage will still be parsed. If there are multiple AirbyteMessage records on the same
+ * line, only the first will be parsed.
+ */
+public class NormalizationAirbyteStreamFactory implements AirbyteStreamFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NormalizationAirbyteStreamFactory.class);
+
+  private final MdcScope.Builder containerLogMdcBuilder;
+  private final Logger logger;
+  private final List<String> dbtErrors = new ArrayList<>();
```

Comment:
I'm not so sure about storing dbtErrors as data within an instance of this object, does that seem fine or is there a better approach?

Reply:
Is the getter use anywhere? creating it in the `create` method and passing it as a parameter will remove any potential side effect.

Reply:
the getter is used [here](https://github.com/airbytehq/airbyte/pull/14790/files#diff-178291195f4aaaf0f6d91905e8ff61dd04655b52067ef259989baf50e24637c4R158)